### PR TITLE
SearchBar.js: Do not open multiple modal windows on multi click

### PR DESCRIPTION
**Pull request by sukhwinder33445**
Related to https://github.com/Icinga/icingaweb2/pull/4538
fix https://github.com/Icinga/icingaweb2/issues/4525

### CODE REVIEW

**Comment by sukhwinder33445** [comment]
When I try to fix without `setTimeout`, the pointer events are not set.

**Review by nilmerg — CHANGES_REQUESTED** [review]
Using `setTimeout` may work, but is not reliable enough. Please use the [`MutationObserver`](https://developer.mozilla.org/de/docs/Web/API/MutationObserver) to solve this.

I would have liked to use just a new event that is triggered by the modal once closed, but that'd require users to upgrade to a new v2.9.4 or so, just to not have the opener being disabled forever. (Without a v2.9.4, tthere's no event that could invalidate the pointer events.)

Use the `MutationObserver` like this:
1) Observe `#layout` and wait for the modal being added to it
2) Observe the modal identified in 1) and wait for the css class `active` to be removed
3) Then reset the pointer events

File: asset/js/widget/SearchBar.js [hunk 1]
```diff
@@ -39,9 +39,28 @@ define(["../notjQuery"], function ($) {
...
             let editorUrl = opener.dataset.searchEditorUrl;
             let filterQueryString = this.filterInput.getQueryString();
+            let layout = document.getElementById('layout');
 
             editorUrl += (editorUrl.indexOf('?') > -1 ? '&' : '?') + filterQueryString;
 
+            // Disable pointer events to block further function calls
+            setTimeout(function () {
+                opener.style.pointerEvents = 'none';
+            }, 20);
```
Comment (nilmerg) [inline asset/js/widget/SearchBar.js:49]: Why not instantly? Or why after 20ms?

File: asset/js/widget/SearchBar.js [hunk 2]
```diff
@@ -39,9 +39,28 @@ define(["../notjQuery"], function ($) {
...
+
+            let observer = new MutationObserver(resetPointerEvents);
+            function resetPointerEvents(mutations) {
```
Comment (nilmerg) [inline asset/js/widget/SearchBar.js:52]: Please don't use a standard function here. It's weird that you reference it already the line beforehand and it's this way bound to the function scope. I don't know if that's reliable enough in terms of garbage collection.

Use a closure (`() => {...}`) instead.

File: asset/js/widget/SearchBar.js [hunk 3]
```diff
@@ -39,9 +39,28 @@ define(["../notjQuery"], function ($) {
...
+                for (let mutation of mutations) {
+                    if (mutation.type === 'childList' && mutation.removedNodes[0]) {
```
Comment (nilmerg) [inline asset/js/widget/SearchBar.js:54]: You don't check if it's the modal that's being removed. Please do so.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: asset/js/widget/SearchBar.js (hunk 1 above)
Comment (sukhwinder33445) [reply asset/js/widget/SearchBar.js:49]: I have not been able to find out why, but it does not work instantly.

Comment (nilmerg) [reply asset/js/widget/SearchBar.js:49]: Then please try `0` for the delay. This way it's right next in the event flow, asap so to say.

Comment (sukhwinder33445) [reply asset/js/widget/SearchBar.js:49]: Tried without `setTimeOut` and works fine now 🤦🏽 (mysterious behaviour).

**Review by nilmerg — APPROVED** [review]

**Pull request merged**